connexion/decorators/security.py

```diff
@@ -252,7 +253,12 @@ def verify_apikey(apikey_info_func, loc, name):
 
     def wrapper(request, required_scopes):
         if loc == 'query':
-            apikey = request.query.get(name)
+            if isinstance(request.query, ImmutableMultiDict):
```

Comment:
I wonder if there's a way to no have to be coupled to werkzeug here.
maybe something like
```
def _immutable_pop(d, kpop):
    """
    takes an immutable dict and the key to pop
    returns a new immutable dict without the popped key, and the value that was popped
    """
    popped = request.query.get(kpop)
    cls = type(d)
    ret = [(k, v) for k, v in d.items() if k != kpop]
    return cls(ret), popped
    
request.query, apikey = _immutable_pop(request.query, name)
```


# copy for immutable_
new = [(k, v) for k, v in request.query.items() if k != name]
request.query = ne
```

Reply:
I'm now thinking that this won't work with MultiDict objects as it will squash the same keys, e.g. ImmutableMultiDict([('a', '1'), ('a', 2), ('a', 3)]) -> ImmutableMultiDict([('a', '1')]). I have updated it with the same try / except handling as used in the uri parsing decorator.